### DISCUSSION TITLE
Disable (grey out) non possible setting combinations

### DISCUSSION THREAD

**Discussion by alexanderniebuhr** [omerfaruk-aran/esphome_samsung_hvac_bus#132] [Questions]
If any preset is chosen, the fan mode doesn't work (which is expected, as this is the same in the smartthings app). However I can still try to set it inside of HA, it will just jump back to "Auto". Same for the swing and windfree. Is there any way to configure HA or the esp, to disable the not valid inputs, e.g. if a preset is chosen, disable fan mode control?

**Comment by bzumik1** [comment]
I agree, I observe same things WindFree should be only possible in cooling mode etc.

**Comment by lanwin** [comment]
@alexanderniebuhr can you please open an Issue. This part is new and was added via pullrequest so I need to look at this.

But I am not sure if the ESPHome Climate API supports that yet. Basically you describe only an what your device is capable of and ESPHome builds the control.

But dont expect a fast fix. I'm pretty busy right now with other stuff.

**Comment by atanasenko** [comment]
A quick glimpse at the esphome climate api doesn't show any way to disable controls. The set of available options and values is calculated in `traits()` method, and it does not seem to depend on actual values being set at that particular point. At least not for HA climate controls.
Maybe there is a `refresh` of a sorts that can be invoked on preset value change to trigger HA to update the ui from new traits?

**Reply by lanwin** [reply]
Meybe it would be a good idea to forward that question directly to ESPHome.
